Sources/ZIPFoundation/Entry.swift

```diff
@@ -59,17 +59,24 @@ public struct Entry: Equatable {
         static let size = 30
         let fileNameData: Data
         let extraFieldData: Data
+        var extraFields: [ExtensibleDataField]?
     }
 
-    struct DataDescriptor: DataSerializable {
+    struct DataDescriptor<T: BinaryInteger>: DataSerializable {
```

Comment:
Nice use of generics. 
As far as I can see, we only use this pattern for `DataDescriptor` - is there a chance to use it for other ZIP/ZIP64 (e.g. EndOfCentralDirectoryRecord)? Or is this impossible due to different layout/members?

Reply:
I think it's hard to use Generic for other structure, the reason is as you said, other ZIP/ZIP64 structures have different layout/members. 😢 Other structures are not so GENERIC.